### PR TITLE
Add logging output for hooks and command

### LINKED ISSUE #38: All commands (especially hooks) should print logs before they start
When there are `pre-connect` and `post-connect` commands that run, the user does not get any indication of what things are running and when. Before each command runs, the program should print relevant log information so the user understands what's running. Child process _outputs_ should generally not be printed, but it's important for users to know what was passed to exec, in case:

* It's slow
* It does something unexpected/malicious

Example output might be:

```
Running pre-connect hook (id=foo): echo foo bar baz
```

Rich text rendering could help this display clearly, e.g. bold/colored text. Or quotes/punctuation.

### PR DESCRIPTION
**Pull request by ryanwholey**
Logs each hook thats run with the hook type, an ID if present, and the description (new field) if present. Logs "no-info" if ID and description are empty. The command arguments could have sensitive output so I opted for description instead. 

image

I also made the error text print red.

fixes #38

### CODE REVIEW

File: internal/command/command.go [hunk 1]
```diff
@@ -11,14 +11,25 @@ import (
...
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
-// Command represents a runnable command.
-type Command struct {
+type commandInput struct {
 	ID          string   `json:"id"`
 	Command     []string `json:"command"`
 	Interactive bool     `json:"interactive"`
+	Name        string   `json:"name"`
```
Comment (ryanwholey) [inline internal/command/command.go:22]:
```suggestion
```

File: internal/command/commands_test.go [hunk 2]
```diff
@@ -28,8 +28,21 @@ func TestParseCommands(t *testing.T) {
...
 		expected := Commands{
-			{ID: "", Command: []string{"echo", "post1"}},
-			{ID: "foo", Command: []string{"echo", "post2"}},
+			{ID: "", Command: []string{"echo", "post1"}, hookType: postConnectHookType},
+			{ID: "foo", Command: []string{"echo", "post2"}, hookType: postConnectHookType},
+		}
+		assert.Equal(t, expected, commands)
+	})
+
+	t.Run("Parse name and description", func(t *testing.T) {
```
Comment (ryanwholey) [inline internal/command/commands_test.go:37]:
```suggestion
	t.Run("Parse", func(t *testing.T) {
```

**Review by bendrucker — CHANGES_REQUESTED** [review]
See other comment re description vs name

Hook types seems unnecessary. The command is not really a hook and that word is getting increasingly overloaded. You have command specific behavior elsewhere too (Interactive). 

Avoid hiding info in structs, stick to passing where practical. For example, pass args Execute that specify what kind of output to return.

**Comment by ryanwholey** [comment]
Sample output with added `sensitive` function:

```sh
> Find RDS instance: aws rds describe-db-instances --db-instance-identifier <id> --query DBInstances[0].DBInstanceIdentifier
> Generate temporary password: aws rds generate-db-auth-token --host <remote-host> --port 5432 --username user
Forwarding from 127.0.0.1:5432 -> 5432
Forwarding from [::1]:5432 -> 5432
> Open client: psql postgres://user:********@localhost:5432/db
Handling connection for 5432
```

File: internal/command/command.go [hunk 3]
```diff
@@ -87,6 +108,13 @@ func (c Command) execute(ctx context.Context, config *Config, args *Args, previo
 		return nil, err
 	}
 
+	cmdStr, err := c.ToString(config, args, outputs)
```
Comment (bendrucker) [inline internal/command/command.go:111]: Avoid ToString as a method name in Go. Either it's a Stringer or the method is doing more than string conversion.

**Review by bendrucker — APPROVED** [review]

File: internal/command/command.go (hunk 3 above)
Comment (ryanwholey) [reply internal/command/command.go:111]: will get it in a follow up

**Pull request merged**